server/src/main/java/org/cloudfoundry/identity/uaa/oauth/AntPathRedirectResolver.java

```diff
@@ -67,6 +69,44 @@ protected boolean redirectMatches(String requestedRedirect, String clientRedirec
         }
     }
 
+    /**
+     * Repeatedly:
+     * <ol>
+     *     <li>checks for a match</li>
+     *     <li>url decodes the requested path</li>
+     * </ol>
+     * until path cannot be url decoded any further. Then normalizes the path before the final check.
+     * <p>
+     *     For example, if example.com/foo is the registered url and example.com/foo/%252e./bar is the requested url,
+     *     checks a match for:
+     *     <ol>
+     *         <li>example.com/foo/%252e./bar</li>
+     *         <li>example.com/foo/%2e./bar</li>
+     *         <li>example.com/foo/../bar</li>
+     *         <li>example.com/bar</li>
+     *     </ol>
+     * </p>
+     */
+    private boolean matches(Predicate<String> matcher, String requestedRedirect) {
```

Comment:
Is there any way to separate this into two functions - one to recursively decode, and another to do the matching?
If not, please rename this function to better reflect the functionality, something like `decodeRecursivelyAndMatch` (just an example, please don't use that one 😄).

Reply:
I'm not sure if it can be split in a way that would enhance readability.
I've renamed the matches() function.